FlixHqMovie/FlixHqMovie/Sences/DetailScreen/MovieDetailViewController.swift

```diff
@@ -0,0 +1,189 @@
+    @IBOutlet private weak var decriptionMovieTextView: UITextView!
+    @IBOutlet private weak var downloadMovieButton: UIButton!
+    @IBOutlet private weak var playMovieButton: UIButton!
+    @IBOutlet private weak var subLabel: UILabel!
+    @IBOutlet private weak var nationalLabel: UILabel!
+    @IBOutlet private weak var ageLabel: UILabel!
+    @IBOutlet private weak var relatedMovieCollectionView: UICollectionView!
+    @IBOutlet private weak var yearLabel: UILabel!
+    @IBOutlet private weak var ibmPointLabel: UILabel!
+    @IBOutlet private weak var movieTitleLabel: UILabel!
+    @IBOutlet private weak var moviePosterImageView: UIImageView!
+    @IBOutlet private weak var castsCollecttionView: UICollectionView!
+    @IBOutlet private weak var loadingActivityIndicator: UIActivityIndicatorView!
+
+    private var playerItem: AVPlayerItem?
+    private var player: AVPlayer?
+    private var currentTime: Float = 0.0
+    private let layerPlayer = AVPlayerLayer()
+    private var playerController = AVPlayerViewController()
+    private var movie: Movie?
+    private var previousTimeWatch: Double = 0.0
+    private var durationMovie: Int = 0
+    private let disposeBag = DisposeBag()
+
+    private let recommendCellSelectedTrigger = PublishSubject<String>()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configView()
+        configPlayVideo()
+        bindingData()
+        configReleatedMovieCollectionView()
+        configCastCollectionView()
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        if let player = playerController.player {
+            let currentTime = Double(CMTimeGetSeconds(player.currentTime()))
+            previousTimeWatch = currentTime
+        }
+    }
+
+    private func configPlayVideo() {
+        playMovieButton.rx.tap.subscribe(onNext: { [unowned self] result in
+            let urlVideo = movie?.sources?[0].url ?? ""
+            let urlSub = movie?.subtitles?[0].url ?? ""
+            guard let url = URL(string: urlVideo) else {
+                return
+            }
+            guard let url2 = URL(string: urlSub) else {
+                return
+            }
+            player = AVPlayer(url: url)
+            let asset = AVAsset(url: url)
+            let subtitleAsset = AVAsset(url: url2)
+            playerItem = AVPlayerItem(asset: asset)
+            player?.replaceCurrentItem(with: playerItem)
+            playerController.player = player
+            playerController.showsTimecodes = true
+
+            if previousTimeWatch > 1 && previousTimeWatch < Double(durationMovie) {
+                playerController.player?.seek(to: CMTime(seconds: previousTimeWatch, preferredTimescale: 1))
+            }
+
+            self.present(playerController, animated: true)
+        })
+        .disposed(by: disposeBag)
+
+    }
+
+    private func configHiddenView(isHiddenView: Bool) {
+        if isHiddenView {
```

Comment:
Mình có thể rút gọn lại còn như thế này nha

```suggestion
       scrollView.isHidden = !isHiddenView
       loadingActivityIndicator.isHidden = isHiddenView
```